bson/_cbsonmodule.c

```diff
@@ -20,6 +20,7 @@
  * should be used to speed up BSON encoding and decoding.
  */
 
+#define PY_SSIZE_T_CLEAN
```

Comment:
See https://github.com/python/cpython/commit/0895793194cd5f1153d2a286ed3a4fc876149b16 and https://docs.python.org/3.8/c-api/arg.html#strings-and-buffers for an explanation of the PY_SSIZE_T_CLEAN changes.

Reply:
[PEP 353 -- Using ssize_t as the index type](https://www.python.org/dev/peps/pep-0353/#conversion-guidelines) is also worth a read. The "Conversion guidelines" section in particular. I've gone ahead and fixed the conversion warnings by using the 3 option: "otherwise, check whether the value fits an int, and raise a ValueError if it doesn't."